gulpfile.babel.js

```diff
@@ -713,93 +281,73 @@ gulp.task('_release', () => {
       'gulpfile.babel.js',
       './util?/**/*'
     ])
-    .pipe(gulp.dest('_release'));
-});
+    .pipe(gulp.dest('release'));
+}
 
-gulp.task('publish:release', ['_release'], () => {
-  return gulp.src('_release')
+function publishRelease(cb) {
+  if(process.env.PUSH !== 'true') {
+    console.log('$PUSH NOT SET TO "true". NOT PUSHING');
+    cb();
+    return;
+  }
+  return gulp.src('release')
     .pipe($.subtree({
       remote: 'origin',
       branch: 'release'
-    }))
-    .pipe(vinylPaths(del));
-});
-
-gulp.task('templates:styles', () => {
-  return gulp.src('templates/**/*.css')
-    .pipe($.autoprefixer(AUTOPREFIXER_BROWSERS))
-    // FIXME: This crashes. It's a bug in gulp-csso,
-    // not csso itself.
-    //.pipe($.csso())
-    .pipe(gulp.dest('dist/templates'));
-});
-
-gulp.task('templates:static', () => {
-  return gulp.src('templates/**/*.html')
-  .pipe($.replace('$$version$$', pkg.version))
-  .pipe($.replace('$$hosted_libs_prefix$$', hostedLibsUrlPrefix))
-  .pipe(gulp.dest('dist/templates'));
-});
-
-// This task can be used if you want to test the templates against locally
-// built version of the MDL libraries.
-gulp.task('templates:localtestingoverride', () => {
-  return gulp.src('templates/**/*.html')
-    .pipe($.replace('$$version$$', '.'))
-    .pipe($.replace('$$hosted_libs_prefix$$', ''))
-    .pipe(gulp.dest('dist/templates'));
-});
-
-gulp.task('templates:images', () => {
-  return gulp.src('templates/*/images/**/*')
-    .pipe($.imagemin({
-      progressive: true,
-      interlaced: true
-    }))
-    .pipe(gulp.dest('dist/templates'));
-});
-
-gulp.task('templates:fonts', () => {
-  return gulp.src('templates/*/fonts/**/*')
-    .pipe(gulp.dest('dist/templates/'));
-});
-
-gulp.task('templates', [
-  'templates:static',
-  'templates:images',
-  'templates:fonts',
-  'templates:styles'
-]);
-
-gulp.task('styles:gen', ['styles'], () => {
-  const MaterialCustomizer = require('./docs/_assets/customizer.js');
-  const templatePath = path.join(__dirname, 'dist', 'material.min.css.template');
-  // TODO: This task needs refactoring once we turn MaterialCustomizer
-  // into a proper Node module.
-  const mc = new MaterialCustomizer();
-  mc.template = fs.readFileSync(templatePath).toString();
-
-  let stream = gulp.src('');
-
-  mc.paletteIndices.forEach(primary => {
-    mc.paletteIndices.forEach(accent => {
-      if (primary === accent) {
-        return;
-      }
+    }));
+}
 
-      if (mc.forbiddenAccents.indexOf(accent) !== -1) {
-        return;
-      }
+gulp.task('styles', gulp.parallel(
+  mdlCss,
+  gulp.series(mdlThemeTemplate, mdlThemes),
+  mdlGrid
+));
+
+gulp.task('scripts', gulp.series(
+  jslint,
+  gulp.parallel(mdlJs, mdlClosureJs)
```

Comment:
Closure should be a completely independent step, otherwise we're forcing MDL developers to have it installed to compile MDL.